### PR TITLE
feat(amplify): source code providers

### LINKED ISSUE #6818: AWS Amplify module with CodeCommit Repository
### The Question

Hello,

With the introduction of the Amplify constructors (App, Branch, Domain), I updated my code from CfnApp to App.
I'm facing a problem with the `accessToken` and `oauthToken` parameters, they are now mandatory if you connect to a repository.

I understand this behavior when you connect to a 3rd party source provider like Github, Gitlab but what about the case of CodeCommit ? 
My deployment used to work with CfnApp connected to a CodeCommit repository and no token as parameters.

Best regards,

### Environment

  - **CDK CLI Version:**  1.30.0
  - **Module Version:** 1.30.0
  - **OS:** OSX Catalina
  - **Language:**  TypeScript

### Other information

Related code from [app.ts](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-amplify/lib/app.ts)
`    if (props.repository && !props.accessToken && !props.oauthToken) {
throw new Error('Either `accessToken` or `oauthToken` must be specified');
    }`

### PR DESCRIPTION
**Pull request by jogold**
### Commit Message

feat(amplify): source code providers

Add source code providers classes for GitHub and CodeCommit that implement
a new `ISourceCodeProvider` interface.

Fixes #6818

BREAKING CHANGE: use the `sourceCodeProvider` prop to connect your app to a source
code provider. The props `repository`, `accessToken` and `oauthToken` do not exist
anymore in `AppProps`.

### End of Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
@MrArnoldPalmer

**Comment by MrArnoldPalmer** [comment]
@skinny85 Would like you check out the additions to codecommit here.

File: packages/@aws-cdk/aws-codecommit/lib/repository.ts [hunk 1]
```diff
@@ -206,6 +227,32 @@ abstract class RepositoryBase extends Resource implements IRepository {
...
+    this.grantPull(grantee);
+    return this.grant(grantee, 'codecommit:GitPush');
+  }
+
+  public grantRead(grantee: iam.IGrantable) {
+    this.grantPull(grantee);
+    return this.grant(grantee,
+      'codecommit:EvaluatePullRequestApprovalRules',
+      'codecommit:Get*',
+      'codecommit:Describe*'
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codecommit/lib/repository.ts]: Trailing comma please 😃.

**Comment by jogold** [comment]
Hey @MrArnoldPalmer, any reason why you're leaving this as `pr/do-not-merge`?

**Comment by MrArnoldPalmer** [comment]
> Hey @MrArnoldPalmer, any reason why you're leaving this as `pr/do-not-merge`?

Nope, sorry just missed it.

**Pull request merged**